pytorch_lightning/overrides/distributed.py

```diff
@@ -73,6 +73,7 @@ def prepare_for_backward(model: DistributedDataParallel, output: Any) -> None:
         # unused parameters. Only if `find_unused_parameters` is set.
         args = list(_find_tensors(output)) if model.find_unused_parameters else []
         reducer = cast(torch._C._distributed_c10d.Reducer, model.reducer)
+        reducer._rebuild_buckets()  # avoids "INTERNAL ASSERT FAILED" with `find_unused_parameters=False`
```

Comment:
I wonder if we should only do this `if not model.find_unused_parameters`

Reply:
I dove into C++land and I don't think it's required as they already check it

https://github.com/pytorch/pytorch/blob/2ba496bc23153a2e227188ebaa08abea52840f63/torch/csrc/distributed/c10d/reducer.cpp#L1631
https://github.com/pytorch/pytorch/blob/2ba496bc23153a2e227188ebaa08abea52840f63/torch/csrc/distributed/c10d/reducer.hpp#L200